tests/functional_small/solvers_nonlinear/updater/basic_backtrack.cc

```diff
@@ -0,0 +1,66 @@
+#include <gtest/gtest.h>
+#include "pressio/solvers_nonlinear.hpp"
+using _this_state_type = Eigen::VectorXd;
+constexpr int _this_N = 3;
+
+struct FakeSolver
+{
+  mutable int count_ = 0;
+  _this_state_type correction_;
+  double residNormCurrCorrStep_ = {};
+
+  FakeSolver() : correction_(_this_N){
+    correction_(0) = 1.;
+    correction_(1) = 2.;
+    correction_(2) = 3.;
+  }
+
+  int count() const { return count_; }
+
+  const _this_state_type & correctionCRef() const{
+    return correction_;
+  }
+
+  double residualNormCurrentCorrectionStep() const{ return 2.5; }
+
+  template<class SystemType>
+  void residualNorm(const SystemType & systemObj,
+		    const _this_state_type & state,
+		    double & residualNorm) const
+  {
+    if (count_ == 0){
+      residualNorm = 3.0;
+    }
+    else if (count_ == 1){
+      residualNorm = 5.0;
+    }
+    else{
+      residualNorm = 1.0;
+    }
+    count_++;
+  }
+
+};
+
+struct DummySystem{};
+
+TEST(solvers_nonlinear, backtracking)
+{
+  pressio::log::initialize(pressio::logto::terminal);
```

Comment:
can you please add a comment here to just say that this is a "simple test to make sure the updater is working correct but without any meaning, the numbers are just crafted to check that teh updater find a direction that is non-increasing the objective"

Reply:
done